src/pages/Pools/hooks/useAddLiquidity.ts

```diff
@@ -0,0 +1,291 @@
+    });
+
+    const result = await CosmJs.execute({
+      address: msg.contractAddress,
+      walletAddr,
+      handleMsg: msg.msg,
+      gasAmount: { denom: ORAI, amount: '0' },
+      funds: msg.funds
+    });
+    console.log('result increase allowance tx hash: ', result);
+
+    if (result) {
+      console.log('in correct result');
+      displayToast(TToastType.TX_SUCCESSFUL, {
+        customLink: `${network.explorer}/txs/${result.transactionHash}`
+      });
+    }
+  };
+
+  const handleAddLiquidity = async (amount1: bigint, amount2: bigint) => {
+    if (!pairInfoData) return;
+    setActionLoading(true);
+    displayToast(TToastType.TX_BROADCASTING);
+
+    try {
+      const oraiAddress = await handleCheckAddress('Oraichain');
+
+      if (token1AllowanceToPair < amount1) {
+        await increaseAllowance('9'.repeat(30), token1!.contractAddress!, oraiAddress);
+        refetchToken1Allowance();
+      }
+      if (token2AllowanceToPair < amount2) {
+        await increaseAllowance('9'.repeat(30), token2!.contractAddress!, oraiAddress);
+        refetchToken2Allowance();
+      }
+
+      // hard copy of from & to token info data to prevent data from changing when calling the function
+      const firstTokenConverts = generateConvertErc20Cw20Message(amounts, token1, oraiAddress);
+      const secTokenConverts = generateConvertErc20Cw20Message(amounts, token2, oraiAddress);
+
+      const msg = generateContractMessages({
+        type: Type.PROVIDE,
+        sender: oraiAddress,
+        fromInfo: token1!,
+        toInfo: token2!,
+        fromAmount: amount1.toString(),
+        toAmount: amount2.toString(),
+        pair: pairInfoData.pairAddr,
+        slippage: (userSlippage / 100).toString()
+      } as ProvideQuery);
+
+      const messages = buildMultipleExecuteMessages(msg, ...firstTokenConverts, ...secTokenConverts);
+
+      const result = await CosmJs.executeMultiple({
+        msgs: messages,
+        walletAddr: oraiAddress,
+        gasAmount: { denom: ORAI, amount: '0' }
+      });
+      console.log('result provide tx hash: ', result);
+
+      if (result) {
+        console.log('in correct result');
+        displayToast(TToastType.TX_SUCCESSFUL, {
+          customLink: `${network.explorer}/txs/${result.transactionHash}`
+        });
+
+        const amountUsdt = Number(toAmount(getUsd(baseAmount, token1, prices) * 2));
+
+        if (typeof onLiquidityChange == 'function') {
+          onLiquidityChange(amountUsdt);
+        }
+      }
+    } catch (error) {
+      console.log('error in providing liquidity: ', error);
+      handleErrorTransaction(error);
+    } finally {
+      setActionLoading(false);
+    }
+  };
+
+  const handleDepositAndStakeAll = async (amount1: bigint, amount2: bigint) => {
+    if (!pairInfoData) return displayToast(TToastType.TX_FAILED, { message: 'Pool information does not exist' });
+
+    setActionAllLoading(true);
+    displayToast(TToastType.TX_BROADCASTING);
+
+    try {
```

Comment:
for discussion, 
should we make a global `try catch` in all `async`? Like wrapping async in a global method to default `try catch` whenever async/await is called. Or keep in individually usage like this?